### PR TITLE
Implement multistep form

### PR DESCRIPTION
**Pull request by abhi886**
# Description

Create a new play called Multistep Form.

Fixes # (788)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Need instructions on tesitng. 
# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by abhi886** [comment]
Hi @atapas,
Could you review my pull request please. 
Thank you.

**Comment by abhi886** [comment]
> Hello @abhi886
> 
> Please pull the latest from the main and follow the readme to fix the formatting and linting errors. Else the build is failing.

Hi @atapas,
I have done the suggested changes. Could you have a look at it please. 
Thank You.

File: src/common/playleaderboard/leaderBoard.css [hunk 1]
```diff
@@ -1,25 +1,25 @@
 .leaderboard-table-cell {
```
Comment (atapas) [inline src/common/playleaderboard/leaderBoard.css:1]: Why this file change is required? Are you using the same formatter?

File: src/plays/react-multistep-form/react-multistep-form.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
+.btn {
```
Comment (atapas) [inline src/plays/react-multistep-form/react-multistep-form.css:1]: Do not override the selectors directly. Add an unique class name at the play details level and use that to prefix all the styeles.

File: src/plays/react-multistep-form/ReactMultistepForm.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      <PersonalInfo {...data} updateField={updateFields} />,
+      <ContactInfo {...data} updateField={updateFields} />,
+      <AccountInfo {...data} updateField={updateFields} />
+    ]);
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+    if (!isLastStep) return next();
+
+    alert(JSON.stringify(data));
```
Comment (atapas) [inline src/plays/react-multistep-form/ReactMultistepForm.tsx:42]: Instead of alert, can we show the output in a styled DIV below the form?

**Review by atapas — CHANGES_REQUESTED** [review]
@abhi886 Great play. Not much to review, most are good. Please take care of a few review comments given and push.

File: src/common/playleaderboard/leaderBoard.css (hunk 1 above)
Comment (abhi886) [reply src/common/playleaderboard/leaderBoard.css:1]: Yes. I am using prettier @2.8.2 as code formatter. I will revert the mentioned changes . Thank you.

File: src/plays/react-multistep-form/ReactMultistepForm.tsx (hunk 3 above)
Comment (abhi886) [reply src/plays/react-multistep-form/ReactMultistepForm.tsx:42]: Sure. I will work on it. Thank you.

**Comment by abhi886** [comment]
Hi @atapas. 
I have made the suggested changes. Could you review it please. 
Thank You.

**Review by atapas — CHANGES_REQUESTED** [review]
@abhi886 Thanks for taking care of the comments. I see the play is missing the play header where we show the play name, creator name etc. Please take a look at other published plays for reference.

image

Actually, when you create plays using the create-react-play command, it creates the base play structure for you including the header. As I do not see it now, you need to check the code of existing plays and add that back.

Please make the changes.

**Comment by atapas** [comment]
The play is responsive but it bit packed when in smaller screen. Some padding around will make it look better.

image

File: src/plays/react-multistep-form/react-multistep-form.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  position: relative;
+  background: white;
+  padding: 1rem;
+  border-radius: 0.5rem;
+  font-family: Arial;
+  max-width: 500px;
+  margin: 0 auto;
+  margin-top: 10%;
+}
+.steps {
```
Comment (atapas) [inline src/plays/react-multistep-form/react-multistep-form.css:47]: You can still make these names unique like `.multistep-play-steps` or add a parent class like `.multistep-play` and use that as prefix to all like `.multistep-play .steps`.

Comment (abhi886) [reply src/plays/react-multistep-form/react-multistep-form.css:47]: Sure. Thank you.

**Comment by abhi886** [comment]
Hi @atapas ,
Are there any more changes that I should do? 
Thank you.

**Review by atapas — APPROVED** [review]
LGTM

**Pull request merged**